internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -171,61 +211,55 @@ func hasCycles(graph *Graph, used map[string]bool, path map[string]bool, currNod
 	return false
 }
 
-func buildDependencyGraph(s convertSidecarOpts) (*Graph, error) {
-	dependencyGraph := Graph{nodes: make(map[string][]string)}
+func buildDependencyGraph(s convertSidecarOpts) (*graph, error) {
+	dependencyGraph := graph{nodes: make(map[string][]string), cycle: make([]string, 0)}
 
-	// sidecar dependencies
+	// Add any sidecar dependencies.
 	for name, sidecar := range s.sidecarConfig {
-		// add any existing dependency to the graph
-		if len(sidecar.DependsOn) != 0 {
-			for dep := range sidecar.DependsOn {
-				// containers being depended on must exist
-				if s.sidecarConfig[dep] == nil && dep != s.workloadName {
-					return nil, errInvalidContainer
-				}
-
-				dependencyGraph.Add(name, dep)
-			}
-		}
-	}
-
-	// add any image dependencies to the graph
-	if len(s.imageConfig.DependsOn) != 0 {
-		for dep := range s.imageConfig.DependsOn {
-			// containers being depended on must exist
+		for dep := range sidecar.DependsOn {
 			if s.sidecarConfig[dep] == nil && dep != s.workloadName {
 				return nil, errInvalidContainer
 			}
 
-			dependencyGraph.Add(s.workloadName, dep)
+			dependencyGraph.Add(name, dep)
+		}
+	}
+
+	// Add any image dependencies.
+	for dep := range s.imageConfig.DependsOn {
+		if s.sidecarConfig[dep] == nil && dep != s.workloadName {
+			return nil, errInvalidContainer
 		}
+
+		dependencyGraph.Add(s.workloadName, dep)
 	}
 
 	return &dependencyGraph, nil
 }
 
-type Graph struct {
+type graph struct {
 	nodes map[string][]string
+	cycle []string
 }
 
-func (graph *Graph) Add(fromNode, toNode string) {
+func (g *graph) Add(fromNode, toNode string) {
 	hasNode := false
 
-	// add origin node if doesn't exist
-	if graph.nodes[fromNode] == nil {
-		graph.nodes[fromNode] = []string{}
+	// Add origin node if doesn't exist.
+	if g.nodes[fromNode] == nil {
```

Comment:
Is this guaranteed to be nil if the key doesn't exist? Is that valid shorthand for:

```go
if _, ok := g.nodes[fromNode]; !ok {
```

Reply:
It appears that the comma ok format and the nil format act similarly in this case, but I adjusted it for consistency with the rest of the codebase.